Document/0x05b-Basic-Security_Testing.md

```diff
@@ -1244,10 +1244,23 @@ For enterprise tools, see the section "Static Source Code Analysis" in the chapt
 
 #### Dynamic Analysis
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1240>
+Dynamic Analysis assesses the application after compiling it and running it, in a similar fashion to black box application testing. The way the code works should be inferred and looked at from a higher level. This allows the tester to properly identify business logic flaws and it helps identify vulnerabilities in live environments. The tester in dynamic analysis can either be an automated tool, such as [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF/), or a real penetration tester. On Android, dynamic analysis can be done by side-loading applications, or by leveraging frameworks such as [Magisk](https://github.com/topjohnwu/Magisk) or [Xposed](https://forum.xda-developers.com/xposed).
```

Comment:
- A tester would normally not compile the app, the target app should be on a release version. The approach to a dynamic analysis is therefore more on the black box side.
- I would mention how dynamic and static analysis relate to and support each other, it's like a feedback loop. Sometimes it might even be a good idea to start with the dynamic analysis to get entry points for the static (reverse eng.).
- I would not say that the tester is an automated tool, rather the tester might use an automated tool..
- You don't need always to side-load the app to perform dyn. ana. the app might be installed already or you need to extra repackage it (for non-rooted) devices.
- I would not say that you use Magisk for dynamic analysis, you use it to root the device, see "###### Rooting with Magisk" also in this chapter.